woocommerce/woocommerce-ios
# Update `RequirementsChecker` to check for free trial plan before checking Woo version

Closes: #10033 


## Description

As reported by Anitaa in p1686893344929769-slack-C03L1NF1EA3, we are currently displaying the Woo version alert when a logged in site is running an expired free trial Woo Express plan, causing the app to be unusable.

This PR fixes the above issue by updating `RequirementsChecker` with the following changes:
- Added a new method to check for WPCom site plan if the site is WPCom.
- Added a new alert to ask for upgrade when a free trial plan expires.
- Updated `StorePickerController` to handle the case when a selected site is running an expired free trial plan.
- Updated `RequirementsChecker` with DI and add unit tests.

Sorry for the large diff, half of the changes are for the unit tests.
## Testing instructions

Prerequisite: Since an Atomic site is still accessible 7 days after its plan's expiration, it's not straightforward to test this case. Make sure that you have tested Woo Express free trial in the past and have at least one site whose plan expired at least 7 days ago.
- With Autoproxxy on, open Store Admin and enter your username.
- You should see a list of sites that you own. Scroll down to the bottom and look for an expired Woo Express site:
<img width="1087" alt="ảnh" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/beee3a7d-7827-43da-96c0-c7bc153ab946">
- Copy the site ID and paste it in `SiteAPIRemote`'s `loadAPIInformation` line 20:

https://github.com/woocommerce/woocommerce-ios/blob/3cddc1f1355c9dbeab530dfdea610f966583af94/Networking/Networking/Remote/SiteAPIRemote.swift#L20

And in `loadSiteCurrentPlan` of `PaymentRemote`: 

https://github.com/woocommerce/woocommerce-ios/blob/3cddc1f1355c9dbeab530dfdea610f966583af94/Networking/Networking/Remote/PaymentRemote.swift#L57

- Build and run the app. You should see an alert saying the site plan is expired. Tapping on Upgrade should navigate to the IAP flow if the feature flag is on, or the web view purchase screen otherwise.
- Dismiss the Upgrade screen and navigate to the site picker. You should see the same alert after the API call is complete.

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/035d989e-5893-46fb-9f22-e4306c528e70" width=320 />
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/450b6273-8500-4f1f-b9d3-fad9a106f3a2" width=320 />


---
